client/src/components/Search.tsx

```diff
@@ -66,7 +67,7 @@ function Search() {
 
   const handleKeyDown = (e: React.KeyboardEvent) => {
     if (e.key === 'Enter') {
-      history.push(`/assets?search=${searchText}`);
+      history.push(`/SearchResults?search=${searchText}`, { category: selectedSearchCategory });
```

Comment:
rather than using location state, i prefer to use the queryParams. the advantage here is that the page is reproducible/bookmarkable i.e. if a user wants to send a link to another user

so unless there is a good reason, let's go with `/SearchResults?search=${searchText}&category=${selectedSearchCategory}`

Reply:
Good call; I haven't thought about bookmarking issue.

Reply:
doesn't look like this was changed, was there a reason you didn't use the url param instead of location state?

Reply:
@jd2rogers2 So for v5 of react-router based on this quick find on StackOverflow ([link](https://stackoverflow.com/questions/35352638/how-to-get-parameter-value-from-query-string)) you need to useLocation on the SearchResults.tsx in order to get the query parameters. Once I migrate to v6, I can actually use a Hook (useParams) which is what I will do in my current sprint.

Note: I did add another parameter to Search.tsx that fills the search term and category on line 70 (      `history.push(/SearchResults?search=${searchText}&category=${selectedSearchCategory});`)

Lastly, I added line 148 (`history.push(`/SearchResults?search=${searchText}&category=${searchCategory}`);`)
Which solves and synchronizes any changes on the /SearchResults component with the URLparams.

v5 before: useHistory [link](https://v5.reactrouter.com/web/api/Hooks/usehistory)
v6 now: useNavigate [link](https://reactrouter.com/en/main/hooks/use-navigate)

v6.4: useParams [link](https://reactrouter.com/en/main/hooks/use-params)
v6.4: UseSearchParams [link](https://reactrouter.com/en/main/hooks/use-search-params)

Let me know if this is satisfactory or if I'm missing the point completely. Apologies for the rewrite